sw-emulator/lib/cpu/src/cpu.rs

```diff
@@ -734,15 +915,19 @@ impl<TBus: Bus> Cpu<TBus> {
         self.global_int_en = false;
 
         self.write_pc(next_pc);
-        println!(
-            "handle_trap: cause={:x}, mtval={:x}, pc={:x}, next_pc={:x}",
-            cause, info, pc, next_pc
-        );
+        // println!(
+        //     "handle_trap: cause={:x}, mtval={:x}, pc={:x}, next_pc={:x}",
+        //     cause, info, pc, next_pc
+        // );
         Ok(())
     }
 
     //// Handle external interrupts
     fn handle_external_int(&mut self, irq: u8) -> StepAction {
+        #[cfg(test)]
+        {
+            self.external_int_counter += 1;
+        }
```

Comment:
There are a couple of places in `cpu.rs` that hardcode memory addresses (`0x5000_0000` and `0x4000_0000`) -- would you mind pulling those out into parameters passed to the CPU?

The MCU is going to use different values for its SRAM shortly.

Reply:
I added a new `CpuArgs` struct with `CpuOrgArgs`. It includes the ROM, ICCM, and DCCM offsets and sizes. I added an implementation of `Default` for `CpuOrgArgs` with the current sizes and offsets to keep things working for now, but the layout can be tweaked.